File: backend/app/schemas/deployments.py
from typing import List

from pydantic import BaseModel


class DeploymentHealthSummary(BaseModel):
    health_status: str
    cpu_percent: int
    memory_percent: int
    last_reported_at: str


class DeploymentEventSummary(BaseModel):
    id: int
    event_type: str
    severity: str
    message: str
    created_at: str


class CustomerDeploymentSummary(BaseModel):
    id: int
    customer_account_id: int
    name: str
    deployment_code: str
    status: str
    version: str
    environment_type: str
    region: str
    launch_url: str
    health_status: str
    last_seen_at: str


class CustomerDeploymentLaunchResponse(BaseModel):
    launch_url: str
    message: str


class PlatformDeploymentListItem(BaseModel):
    id: int
    customer_account_id: int
    customer_account_name: str
    name: str
    deployment_code: str
    status: str
    version: str
    environment_type: str
    region: str
    health_status: str
    last_seen_at: str


class PlatformDeploymentDetail(BaseModel):
    id: int
    customer_account_id: int
    customer_account_name: str
    name: str
    deployment_code: str
    status: str
    version: str
    environment_type: str
    region: str
    launch_url: str
    health_status: str
    last_seen_at: str
    events: List[DeploymentEventSummary]

File: Titanfrontend/src/types/deployments.ts
export type PlatformDeployment = {
    id: number;
    customer_account_id: number;
    customer_account_name: string;
    name: string;
    deployment_code: string;
    status: string;
    version: string;
    environment_type: string;
    region: string;
    health_status: string;
    last_seen_at: string;
};

export type DeploymentEventSummary = {
    id: number;
    event_type: string;
    severity: string;
    message: string;
    created_at: string;
};

export type PlatformDeploymentDetail = {
    id: number;
    customer_account_id: number;
    customer_account_name: string;
    name: string;
    deployment_code: string;
    status: string;
    version: string;
    environment_type: string;
    region: string;
    launch_url: string;
    health_status: string;
    last_seen_at: string;
    events: DeploymentEventSummary[];
};